apps/full-site-editing/full-site-editing-plugin/global-styles/src/dom-updater.js

```diff
@@ -14,23 +15,33 @@ export default ( options, getOptionValue ) => {
 	domReady( () => {
 		const current = {};
 		const cssVariables = {};
+		const styleElement = document.createElement( 'style' );
+		document.body.appendChild( styleElement );
+		const styleSheet = styleElement.sheet;
+
 		options.forEach( option => {
 			current[ option ] = null;
 			cssVariables[ option ] = `--${ option.replace( '_', '-' ) }`;
 		} );
 
 		subscribe( () => {
+			let styleProps = '';
 			Object.keys( current ).forEach( key => {
 				const value = getOptionValue( key );
-				if ( current[ key ] !== value ) {
+				if ( ! isEmpty( value ) && current[ key ] !== value ) {
 					current[ key ] = value;
-					// We want to scope this to the root node of the editor.
-					const node = document.getElementsByClassName( 'editor-styles-wrapper' )[ 0 ];
-					if ( node ) {
-						node.style.setProperty( cssVariables[ key ], value );
-					}
+					styleProps += `${ cssVariables[ key ] }:${ value };`;
 				}
 			} );
+			// We want to scope this to the root node of the editor.
+			// We need this to be a stylesheet rather than inline styles
+			// so the styles apply to all editor instances incl. previews.
+			if ( ! isEmpty( styleProps ) ) {
+				styleSheet.insertRule(
```

Comment:
This is appending new rules every time it has to add something. Could we delete the existing rule first and then add the new one?

The existing properties for font-families can't be null, but we don't know about the future rules. So, if we don't fix this, we could end with a very subtle bug such as:

```
// rule 1 is added
.editor-styles-wrapper {
  some-prop: some-value,
  other-prop: other-value
}
// value changes and rule 2 is added
.editor-styles-wrapper {
  other-prop: other-value;
}
```

Note how despite rule 2 doesn't bind a value for `some-prop` there is still one due to the cascade!

Reply:
Also, perhaps I'm doing things wrong, but I don't see the rulesets within the added DOM node: its content is void. Is this something we could do differently? The above error would have been difficult to debug if we had to inspect the raw CSS object model.

Reply:
Similar to my comment above, if I'm observing the behavior correctly, then the implementation I found before adding my code seems to set the value back to the default rather than removing it.

 So, in your example case above, what would happen is 

```
// rule 1 is added
.editor-styles-wrapper {
  some-prop: new-value,
  other-prop: new-value
}

// rule 2 is added
// only one prop is changed, the other is still valid because of the cascade
.editor-styles-wrapper {
  other-prop: other-value;
}

// rule 3 is added
// one prop is changed, the other set to the default value
.editor-styles-wrapper {
  some-prop: default-value
  other-prop: yet-another-value;
}
```

Reply:
> Also, perhaps I'm doing things wrong, but I don't see the rulesets within the added DOM node: its content is void.

Yes, that is because Chrome doesn't support editing of CSSOM-based rules in DevTools:
https://bugs.chromium.org/p/chromium/issues/detail?id=387952

Reply:
As per discussion on Slack changed the implementation to only ever use a single selector as well as moving away from CSSOM for easier debugging.